### PR TITLE
Add listLoanTokens rpc

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Add listLoanTokens rpc

#### Which issue(s) does this PR fixes?:

Fixes part of #488

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 1]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
   interest?: BigNumber
 }
 
+export interface ListLoanTokenData {
+  [key: string]: ListLoanTokenDetail
+}
+
+export interface ListLoanTokenDetail {
+  token: TokenData
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:246]:
```suggestion
  token: LoanTokenData
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 2]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
...
+  priceFeedId: string
+  interest: BigNumber
+}
+
+export interface TokenData {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:251]:
```suggestion
export interface LoanTokenData {
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 3]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
...
+  [key: string]: TokenDetail
+}
+
+export interface TokenDetail {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:255]:
```suggestion
export interface LoanTokenDetail {
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 4]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
   interest?: BigNumber
 }
 
+export interface ListLoanTokenData {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:241]:
```suggestion
export interface ListLoanTokenResult {
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 5]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
...
+  [key: string]: ListLoanTokenDetail
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:242]:
```suggestion
  [key: string]: ListLoanTokenInfo
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 6]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
...
+}
+
+export interface ListLoanTokenDetail {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:245]:
```suggestion
export interface LoanTokenDetail {
```

File: packages/jellyfish-api-core/src/category/loan.ts (hunk 1 above)
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:246]:
```suggestion
  token: LoanTokenInfo
```

File: packages/jellyfish-api-core/src/category/loan.ts (hunk 3 above)
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:255]:
```suggestion
export interface LoanTokenInfo {
```

File: packages/jellyfish-api-core/src/category/loan.ts (hunk 2 above)
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:251]:
```suggestion
export interface LoanTokenData {
```

File: packages/jellyfish-api-core/src/category/loan.ts (hunk 3 above)
Comment (canonbrother) [reply packages/jellyfish-api-core/src/category/loan.ts:255]: 1. follow our naming standard here
`result { info }`

2. prefix the `Loan` for better differentiate them if we don't refer to `Token` interface

**Review by canonbrother — APPROVED** [review]

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 7]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
   interest?: BigNumber
 }
 
+export interface ListLoanTokenResult {
+  [key: string]: ListLoanDetail
+}
+
+export interface ListLoanDetail {
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/loan.ts:245]:
```suggestion
export interface LoanTokenDetails {
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 8]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
...
+}
+
+export interface LoanTokenInfo {
+  collateralAddress: string
+  creationHeight: BigNumber
+  creationTx: string
+  decimal: BigNumber
+  destructionHeight: BigNumber
+  destructionTx: string
+  finalized: false
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/loan.ts:262]: do we need to hardcode `false` to the interface?
```suggestion
  finalized: boolean
```

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/loan.ts:262]: Nope LOL

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 9]
```diff
@@ -229,6 +238,40 @@ export interface SetLoanToken {
...
+  isLPS: boolean
+  isLoanToken: boolean
+  limit: BigNumber
+  mintable: boolean
+  minted: BigNumber
+  name: string
+  symbol: string
+  symbolKey: string
+  tradeable: boolean
+}
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/loan.ts:273]:
```suggestion
  token: token.TokenResult
  priceFeedId: string
  interest: BigNumber
}
```

Comment (surangap) [inline packages/jellyfish-api-core/src/category/loan.ts:273]:
```suggestion
export interface LoanTokenDetails {
  token: token.TokenResult
  priceFeedId: string
  interest: BigNumber
}
```

Comment (surangap) [reply packages/jellyfish-api-core/src/category/loan.ts:273]: the above suggestion is to use the already existing structure in  `token.ts`. so you may have to refer to it. 
Also, i noticed that `isLoanToken: boolean` is missing from that one. newly added filed. Could you update that one and use here?

File: packages/jellyfish-api-core/src/category/token.ts [hunk 10]
```diff
@@ -114,6 +114,7 @@ export interface TokenInfo {
   tradeable: boolean
   isDAT: boolean
   isLPS: boolean
+  isLoanToken?: boolean
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/token.ts:117]:
```suggestion
  isLoanToken: boolean
```

**Review by surangap — APPROVED** [review]

**Review by Jouzo — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**